hack/autodoc/autodoc.go

```diff
@@ -0,0 +1,347 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+const (
+	spaceMargin             = 10
+	errorExitCode           = 1
+	okExitCode              = 0
+	expectedDataChunksCount = 3
+	defaultMdElementSize    = 10
+	conditionType           = "Ready"
+)
+
+type reasonMetadata struct {
+	groupOrder      int
+	crState         string
+	conditionType   string
+	conditionStatus bool
+	conditionReason string
+	remark          string
+}
+
+func main() {
+	dataForProcessing := extractData()
+	dataChunks := strings.Split(dataForProcessing, "====")
+	if len(dataChunks) != expectedDataChunksCount {
+		fmt.Println("'extract_conditions_data.sh' data output failed, it should contain 3 elements")
+		os.Exit(errorExitCode)
+	}
+
+	constReasons := getConstReasons(dataChunks[0])
+
+	errors, reasonsMetadata := getAndValidateReasonsMetadata(dataChunks[1])
+	if len(errors) > 0 {
+		printErrors(errors)
+		os.Exit(errorExitCode)
+	}
+
+	errors = checkIfConstsAndMetadataAreInSync(constReasons, reasonsMetadata)
+	if len(errors) > 0 {
+		fmt.Println("The declared reasons in const Go section are out out sync with Reasons metadata")
+		printErrors(errors)
+		os.Exit(errorExitCode)
+	}
+
+	mdTableContent := mdTableToStruct(dataChunks[2])
+	errors = compareContent(mdTableContent, reasonsMetadata)
+	if len(errors) > 0 {
+		printErrors(errors)
+		fmt.Println("Below can be found auto-generated table which contain new changes:")
+		fmt.Println(buildMdTable(reasonsMetadata))
+		os.Exit(errorExitCode)
+	}
+
+	os.Exit(okExitCode)
+}
+
+func extractData() string {
+	cmd := exec.Command("/bin/sh", "hack/autodoc/extract_conditions_data.sh")
```

Comment:
extract script name (you use it in main in message) to make it easier to change.